sunpy/io/special/srs.py

```diff
@@ -0,0 +1,203 @@
+"""
+This module implements SRS File Reader.
+"""
+import datetime
+from collections import OrderedDict
+
+import numpy as np
+from astropy.table import QTable, MaskedColumn, Column, vstack
+import astropy.io.ascii
+import astropy.units as u
+
+__all__ = ['read_srs']
+
+
+def read_srs(filepath):
+    """
+    Parse a SRS table from NOAA SWPC.
+
+    Parameters
+    ----------
+    filepath : `str`
+        The full path to a SRS table.
+
+    Returns
+    -------
+
+    table : `astropy.table.QTable`
+        Table containing a stacked table from all the tables in the SRS file.
+        The header information is stored in the ``.meta`` attribute.
+
+    """
+    with open(filepath) as srs:
+        file_lines = srs.readlines()
+
+    header, section_lines = split_lines(file_lines)
+
+    return make_table(header, section_lines)
+
+
+def make_table(header, section_lines):
+    """
+    From the seperated section lines and the header, clean up the data and
+    convert to a QTable.
+    """
+    meta_data = get_meta_data(header)
+
+    tables = []
+    for i, lines in enumerate(section_lines):
+        if lines:
+            key = list(meta_data['id'].keys())[i]
+            t1 = astropy.io.ascii.read(lines)
+            t1.add_column(Column(data=[key] * len(t1), name="ID"), index=0)
+            tables.append(t1)
+
+    out_table = vstack(tables)
+
+    # Parse the Location column in Table 1
+    if 'Location' in out_table.columns:
+        col_lat, col_lon = parse_location(out_table['Location'])
+        del out_table['Location']
+        out_table.add_column(col_lat)
+        out_table.add_column(col_lon)
+
+    # Parse the Lat column in Table 3
+    if 'Lat' in out_table.columns:
+        parse_lat_col(out_table['Lat'], out_table['Latitude'])
+        del out_table['Lat']
+
+    # Give columns more sensible names
+    out_table.rename_column("Nmbr", "Number")
+    out_table.rename_column("NN", "Number of Sunspots")
+    out_table.rename_column("Lo", "Carrington Longitude")
+    out_table.rename_column("MagType", "Mag Type")
+    out_table.rename_column("LL", "Longitudinal Extent")
+
+    # Define a Solar Hemispere Unit
+    a = {}
+    u.def_unit(
+        "SH",
+        represents=(2 * np.pi * u.solRad**2),
+        prefixes=True,
+        namespace=a,
+        doc="A solar hemisphere is the area of the visible solar disk.")
```

Comment:
Perhaps we should have a units submodule? Any suggestions of other units we might want?